ister.py

```diff
@@ -1012,6 +1009,10 @@ def validate_version_template(template):
 def validate_softmgr_template(template):
     """Attempt to verify the package manager is sane
     """
+    # Supply default SoftwareManager value if not defined for backwards compatibility
```

Comment:
So the placement here I agree is a little confusing but as discussed in the PR where this was added it does make it easier to avoid modifying the template after it was read it (I realize we do have an exception for this in the code but that can't be done early unlike this one). Perhaps if we get more cases like this a set defaults function would make sense but for now I'm okay with where this code was.

Reply:
ok, we can improve this in the future

Reply:
Um I don't see why this part of the change needs to be kept. The explanation in the git commit doesn't make sense to me either as ister is imported successfully already in the gui and test scripts.

Reply:
the changes to encryption needs to keep the password hide, if ister.py is called in a subprocess this will be store in the template.json in the file system, the way to avoid this was change how ister_gui.py calls ister.py instead use a subprocess ister.py is used as python module and its pass the template dict directly to its functions, https://github.com/bryteise/ister/pull/125 the previous place where was this default assignation is where the json is being validated and that function is executed just when the ister.py is called directly in the command line

Reply:
I think the correct change here is to update ister_gui.py to include the SoftwareManager in the template then not move where the default is added. That can be done in another commit so you don't need to rebase.

Reply:
if it is an optional parameter, even if it is not defined it most be assigned, if it is not moved you force to define this option when ister is used as a module

Reply:
My thinking is that install_os requires a valid in memory template given to it. A template file can have an optional component but not that isn't what install_os takes as an argument. I don't want to have any function in the module able to handle an incomplete template.

Reply:
ok, changing it

Reply:
Ready